keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,295 @@
+az keyvault set-policy --name <keyvault-name> --spn <your-service-principal-id> --secret-permissions <secret-permissions>
+```
+
+#### Create Client
+
+In order to interact with the Azure Key Vault Secrets service you'll need to create an instance of the Secret Client class. To do this you'll need the application id and application key of an application in Azure Active Directory authorized with access to key vault.
+
+Once you have the values of the application id, application key and tenant id you can create the secret client.
+The following environment varaibles need to be configured to authorize with your key vault via default credentials.
+1. AZURE_CLIENT_ID - The application id.
+2. AZURE_CLIENT_KEY - The application key.
+3. AZURE_TENANT_ID - The id of the Azure Active Directory under which your application is registered.
+
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+or
+
+```Java
+SecretAsyncClient client = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+## Key concepts
+
+### Secret
+
+A secret is the fundamental resource within an Azure Key Vault. In its simplest form it is a name and a value. However, there are additional properties such as:
+   1. content type : Type of the secret value such as a password.
+   2. tags : Application specific metadata in the form of key-value pairs.
+   3. expires : Specifies a UTC time at which secret will no loger be active.
+   4. notBefore : Specifies a UTC time after which secret will be active.
+
+### Secret Client
+
+The client performs the interactions with the Azure Key Vault service, getting, setting, updating, deleting, and listing secrets. An asynchronous, `SecretAsyncClient`, and synchronous, `SecretClient`, client exists in the SDK allowing for selection of a client based on an application's use case.
+
+## Sync API Examples
+
+The following sections provide several code snippets covering some of the most common Azure Key Vault Secret Service tasks, including:
+- [Create a Secret](#create-a-Secret)
+- [Retrieve a Secret](#retrieve-a-Secret)
+- [Update an existing Secret](#update-an-existing-Secret)
+- [Delete a Secret](#delete-a-Secret)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault.
+- setSecret creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+Secret secret = secretClient.setSecret("secret_name", "secret_value").value();
+System.out.printf("Secret is created with name %s and value %s \n", secret.name(), secret.value());
+
+```
+
+### Retrieve a Secret
```

Comment:
I'd add in-code comments both here and elsewhere - for example when you're adding a secret first to change it later or doing things like changing the expiry time